### PR TITLE
updated dependencies and posix expansion for values

### PR DESCRIPTION
**Pull request by cviecco**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/Cloud-Foundations/keymaster
 
-go 1.18
+go 1.21
```
Comment (rgooch) [inline go.mod:3]: Do we need to force the most recent version of Go? Is 1.19 sufficient?

Comment (cviecco) [reply go.mod:3]: unfortunatelly, the library I chose requires 1.21 (1.22 is the lastest)

File: README.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ Pre-build binaries (both RPM and DEB) can be found here: [releases page](https:/
 ### Building from Source
 
 #### Prerequisites
-* go >= 1.13
+* go >= 1.19
```
Comment (rgooch) [inline README.md:34]: Should this be 1.21 to be consistent?

File: README.md [hunk 3]
```diff
@@ -79,6 +80,9 @@ Keymaster supports SQLite and PostgreSQL to store u2f tokens or username and pas
 To use keymasterd as an openid connect IDP please consult the documents
 [here](docs/website/openidc-idp.md)
 
+##### SSH Cerfificate exteansion expansion
+Some systems like github.com allow the use of ssh certificates to authenticate users. To do so it is required to have speficic extensions in the ssh certificate. To accomodate this we have a bash like extension mechanism for expanding the username (some deployments require prefixes and some require some character subsituttions). We use posix expression expanding system, but we allow reserver the pipe "|" so that we can do some future expansions.
+
```
Comment (rgooch) [inline README.md:85]: s/allow/also/ ?
s/reserver/reserved/ ?

File: README.md (hunk 2 above)
Comment (cviecco) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (cviecco) [reply README.md:86]: fixed

**Review by rgooch — APPROVED** [review]

**Pull request merged**